python/pacemaker/_cts/cib.py

```diff
@@ -72,7 +72,14 @@ def new_ip(self, name=None):
 
         if ":" in ip:
             r["cidr_netmask"] = "64"
-            r["nic"] = self._cm.env["nic"]
+
+            if self._cm.env["nic"] is not None:
+                r["nic"] = self._cm.env["nic"]
```

Comment:
How about the following? The idea is to set `r["nic"]` to the env value first, and then if it's still unset, default it to `"eth0"` if appropriate.
```
            r["nic"] = self._cm_env["nic"]
            if r["nic"] is None and ip.lstrip().startswith("fe80::"):
                r["nic"] = "eth0"
```

I don't remember much about how IPv6 works. Are we sure we want to set this default _only_ for link-local addresses?

I was going to suggest removing the `lstrip()`, but I saw that in some places we default `IPBase` to a string that begins with a space. I have no idea why this was done. It goes back to ff6fbb10 and e599838b. We certainly don't need to worry about that in this PR -- it's easier to use `lstrip()` here as you've done.

Reply:
AFAICS for non-link-local IPv6, IPaddr2 RA tries to find a suitable nic from the route table as well. So IMO this should be the best we can do when this `--nic` option is omitted.